### PR TITLE
Add flag to `mc replicate` for existing object replication

### PR DESCRIPTION
**Pull request by poornas**
Needs https://github.com/minio/minio-go/pull/1483/files and https://github.com/minio/madmin-go/pull/6
For https://github.com/minio/minio/pull/12109

1.  for existing object replication
`mc replicate add|edit --replicate "existing-objects,.."`

2. for replication reset
`mc replicate resync alias/bucket`

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

**Comment by vadmeste** [comment]
@poornas could you fix conflicts here ?

File: cmd/replicate-add.go [hunk 1]
```diff
@@ -173,24 +183,30 @@ func mainReplicateAdd(cliCtx *cli.Context) error {
...
+		RuleStatus:              ruleStatus,
+		ID:                      cliCtx.String("id"),
+		DestBucket:              cliCtx.String("remote-bucket"),
+		Op:                      replication.AddOption,
+		ReplicateDeleteMarkers:  dmReplicateStatus,
+		ReplicateDeletes:        deleteReplicationStatus,
+		ReplicaSync:             replicaSync,
+		ExistingObjectReplicate: existingReplicationStatus,
 	}
+	fmt.Println("existing stat>", existingReplicationStatus)
```
Comment (vadmeste) [inline cmd/replicate-add.go:209]: Not needed

File: cmd/replicate-reset.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+var replicateResetFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "older-than",
+		Usage: "re-replicate objects older than n days",
+	},
+}
+
+var replicateResetCmd = cli.Command{
+	Name:         "reset",
```
Comment (vadmeste) [inline cmd/replicate-reset.go:42]: Maybe "resync" is a better name ?

**Review by vadmeste — COMMENTED** [review]
Some comments

File: cmd/replicate-reset.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/replicate-reset.go:42]: yeah resync is better word than `reset`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/replicate-reset.go (hunk 2 above)
Comment (poornas) [reply cmd/replicate-reset.go]: changed this

File: cmd/replicate-resync.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+func (r replicateResyncMessage) JSON() string {
+	r.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(r, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(jsonMessageBytes)
+}
+
+func (r replicateResyncMessage) String() string {
+	return console.Colorize("replicateResyncMessage", fmt.Sprintf("Replication reset started for %s with ID %s", r.URL, r.ResetID))
```
Comment (kannappanr) [inline cmd/replicate-resync.go:88]: `Replication reset started for` -> `Replication resync started for`

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/admin-bucket-remote-add.go [hunk 4]
```diff
@@ -138,6 +138,8 @@ type RemoteMessage struct {
 	ReplicationSync     bool          `json:"replicationSync"`
 	Proxy               bool          `json:"proxy"`
 	HealthCheckDuration time.Duration `json:"healthcheckDuration"`
+	ResetID             string        `json:"resetID"`
+	ResetBefore         time.Time     `json:"resetBeforeDate"`
```
Comment (harshavardhana) [inline cmd/admin-bucket-remote-add.go:142]: If we are going to call it resync, let's rename everything properly

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/auto-complete.go [hunk 5]
```diff
@@ -272,6 +272,7 @@ var completeCmds = map[string]complete.Predictor{
 	"/replicate/export": s3Complete{deepLevel: 2},
 	"/replicate/import": s3Complete{deepLevel: 2},
 	"/replicate/status": s3Complete{deepLevel: 2},
+	"/replicate/reset":  s3Complete{deepLevel: 2},
```
Comment (harshavardhana) [inline cmd/auto-complete.go:275]:
```suggestion
	"/replicate/resync":  s3Complete{deepLevel: 2},
```

File: cmd/replicate-reset.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
```
Comment (harshavardhana) [inline cmd/replicate-reset.go:42]:
```suggestion
	Name:         "resync",
```

File: cmd/replicate-reset.go [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+	Usage:        "reset replication on previously replicated objects",
+	Aliases:      []string{"resync"},
```
Comment (harshavardhana) [inline cmd/replicate-reset.go:44]:
```suggestion
	Aliases:      []string{"reset"},
```

File: cmd/replicate-reset.go [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+var replicateResetFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "older-than",
+		Usage: "re-replicate objects older than n days",
+	},
+}
+
+var replicateResetCmd = cli.Command{
+	Name:         "reset",
+	Usage:        "reset replication on previously replicated objects",
```
Comment (harshavardhana) [inline cmd/replicate-reset.go:43]:
```suggestion
	Usage:        "re-replicate all previously replicated objects",
```

File: cmd/replicate-reset.go [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+   {{.HelpName}} TARGET
+
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
+  1. Re-replicate previously replicated objects in bucket "mybucket" for alias "myminio".
+	   {{.Prompt}} {{.HelpName}} myminio/mybucket
+
+  2. Reset server side replication for bucket "mybucket" for alias "myminio" for objects older than 60 days.
```
Comment (harshavardhana) [inline cmd/replicate-reset.go:62]:
```suggestion
  2. Re-replicate all objects older than 60 days in "mybucket".
```

File: cmd/replicate-reset.go [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+
+USAGE:
+   {{.HelpName}} TARGET
+
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
```
Comment (harshavardhana) [inline cmd/replicate-reset.go:58]: there is some tabbing issue here, looks like indentation is off from other commands

File: go.sum [hunk 11]
```diff
@@ -249,14 +249,21 @@ github.com/minio/colorjson v1.0.1 h1:+hvfP8C1iMB95AT+ZFDRE+Knn9QPd9lg0CRJY9DRpos
...
 github.com/minio/md5-simd v1.1.1 h1:9ojcLbuZ4gXbB2sX53MKn8JUZ0sB/2wfwsEcRw+I08U=
 github.com/minio/md5-simd v1.1.1/go.mod h1:XpBqgZULrMYD3R+M28PcmP0CkI7PEMzB3U77ZrKZ0Gw=
 github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78/go.mod h1:mTh2uJuAbEqdhMVl6CMIIZLUeiMiWtJR4JB8/5g2skw=
 github.com/minio/minio-go/v7 v7.0.11-0.20210517200026-f0518ca447d6 h1:GVR+UTvfe2r2YTYHWrA/yRF5nouMjJh3kwxNTZ8npso=
 github.com/minio/minio-go/v7 v7.0.11-0.20210517200026-f0518ca447d6/go.mod h1:td4gW1ldOsj1PbSNS+WYK43j+P1XVhX/8W8awaYlBFo=
+github.com/minio/minio-go/v7 v7.0.11-0.20210602150421-c1d2e1ce60f0 h1:B47X2+bmKGXaaj3UcNW3pF7whYktOfLGgMRWLnn7ah4=
+github.com/minio/minio-go/v7 v7.0.11-0.20210602150421-c1d2e1ce60f0/go.mod h1:td4gW1ldOsj1PbSNS+WYK43j+P1XVhX/8W8awaYlBFo=
+github.com/minio/minio-go/v7 v7.0.11-0.20210607181445-e162fdb8e584 h1:GgOqN/DE44zZh3+sg5elR6diWq+Qv4dN7U2KvPVfe1I=
+github.com/minio/minio-go/v7 v7.0.11-0.20210607181445-e162fdb8e584/go.mod h1:WoyW+ySKAKjY98B9+7ZbI8z8S3jaxaisdcvj9TGlazA=
 github.com/minio/pkg v1.0.3 h1:tUhM6lG/BdNB0+5f2RbE4ifCAYwMs6cRJnZ/AY0WIeQ=
```
Comment (harshavardhana) [inline go.sum:267]: run `go mod tidy`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**